display_list/skia/dl_sk_canvas.cc

```diff
@@ -49,6 +47,19 @@ static SkPaint ToSk(const DlPaint& paint, bool force_stroke = false) {
     color_filter = invert_filter;
   }
   sk_paint.setColorFilter(color_filter);
+
+  // On the Impeller backend, we will only support dithering of *gradients*,
+  // and it will be enabled by default (without the option to disable it). Until
+  // Skia support is completely removed, we only want to respect the dither flag
+  // for gradients (otherwise it will also apply to, for example, images, which
+  // is not supported in Impeller).
+  //
+  // See https://github.com/flutter/flutter/issues/112498.
+  auto color_source = paint.getColorSourcePtr();
```

Comment:
Combine this with the code up above that calls setShader. No need to get the ColorSourcePtr twice...

Reply:
Done.